frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/licenses/LicensesPortlet.java

```diff
@@ -405,46 +410,55 @@ public void addObligations(ActionRequest request, ActionResponse response) throw
 
-
-        Obligation oblig = new Obligation();
-        //add temporary id
-        oblig.setId(TMP_TODO_ID_PREFIX + UUID.randomUUID().toString());
-        if (obligationIds != null) {
-            for (String obligationId : obligationIds) {
-                if (obligationId != null && !obligationId.isEmpty()) {
-                    oblig.addToObligationDatabaseIds(obligationId);
-                }
-            }
-        } else {
-            oblig.setObligationDatabaseIds(Collections.emptySet());
-        }
-        oblig.setText(obligsText);
-        oblig.setTitle(StringUtils.EMPTY);
-
         User user = UserCacheHolder.getUserFromRequest(request);
         String moderationComment = request.getParameter(PortalConstants.MODERATION_REQUEST_COMMENT);
         if(moderationComment != null) {
             user.setCommentMadeDuringModerationRequest(moderationComment);
         }
 
-        oblig.addToWhitelist(user.getDepartment());
-
-        if (bools != null) {
-            List<String> theBools = Arrays.asList(bools);
-            oblig.setDevelopment(theBools.contains("development"));
-            oblig.setDistribution(theBools.contains("distribution"));
-        } else {
-            oblig.setDevelopment(false);
-            oblig.setDistribution(false);
-        }
         try {
             LicenseService.Iface client = thriftClients.makeLicenseClient();
-            RequestStatus requestStatus = client.addObligationsToLicense(oblig, licenseID, user);
-
-            setSessionMessage(request, requestStatus, "License", "update");
-
+            if (null == obligationIds) {
+                if (CommonUtils.isNotNullEmptyOrWhitespace(obligsText)) {
+                    Obligation obl = new Obligation();
+                    obl.setId(TMP_OBLIGATION_ID_PREFIX + UUID.randomUUID().toString());
+                    obl.setText(obligsText);
+                    obl.addToWhitelist(user.getDepartment());
+                    obl.setTitle(obligTitle);
+                    obl.setObligationLevel(ObligationLevel.LICENSE_OBLIGATION);
+                    if (bools != null) {
+                        List<String> theBools = Arrays.asList(bools);
+                        obl.setDevelopment(theBools.contains("development"));
+                        obl.setDistribution(theBools.contains("distribution"));
+                    } else {
+                        obl.setDevelopment(false);
+                        obl.setDistribution(false);
+                    }
+                    RequestStatus requestStatus = client.addObligationsToLicense(obl, licenseID, user);
+                    setSessionMessage(request, requestStatus, "License", "update");
+                }
+            } else {
+                List<Obligation> obls = client.getObligationsByIds(Arrays.asList(obligationIds));
+                obls.stream().filter(Objects::nonNull).forEach(obl -> wrapException(() -> {
+                    Obligation oblCopy = obl.deepCopy();
```

Comment:
that seems to be the only relevant part for the if-else

Reply:
refactored the code.